### PR TITLE
Added a Thread Safe Option for Retrieving from the Handler Cache

### PR DESCRIPTION
**Pull request by MinuraPunchihewa**
## Description

This PR adds a thread safe option for retrieving handlers from the handler cache.

## Type of change

- [X] 🐛 Bug fix (non-breaking change which fixes an issue)

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Checklist:

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by MinuraPunchihewa** [comment]
Hey @StpMax,
Do you think that we should also allow items to be deleted from the cache in a thread safe manner? I think it is possible to do this, but do you see any negative implications?

File: mindsdb/interfaces/database/integrations.py [hunk 1]
```diff
@@ -94,19 +94,31 @@ def set(self, handler: DatabaseHandler):
                 pass
             self._start_clean()
 
-    def get(self, name: str) -> Optional[DatabaseHandler]:
+    def get(self, name: str, thread_safe: bool = False) -> Optional[DatabaseHandler]:
```
Comment (StpMax) [inline mindsdb/interfaces/database/integrations.py:97]: I don't really like to add additional arguments to functions. Better keep interface simple. What if add 'thread_safe' arg to the handler class itself?
Therefore 'set' may look like:
```python
def set(self, handler: DatabaseHandler):
    ...
    key = (handler.name, ctx.company_id, 0 if getattr(handler, 'thread_safe', False) else thread_id)
    ...
```
and 'get' like:
```python
def get(self, name: str) -> Optional[DatabaseHandler]:
    ...
    key = (name, ctx.company_id, thread_id)
    if key not in cache:
        key = (name, ctx.company_id, 0)
    if (
        key not in self.handlers
        or self.handlers[key]['expired_at'] < time()
    ):
        return None
    ...
```
what do you think?

Comment (MinuraPunchihewa) [reply mindsdb/interfaces/database/integrations.py:97]: @StpMax This makes sense. I will push the suggested changes shortly.

**Comment by MinuraPunchihewa** [comment]
Closing in favor of https://github.com/mindsdb/mindsdb/pull/9517

**Comment by MinuraPunchihewa** [comment]
@StpMax I created a new PR with the suggested changes [here](https://github.com/mindsdb/mindsdb/pull/9517).

**Pull request closed without merging**